pkg/cluster/fixssh.go

```diff
@@ -49,11 +49,26 @@ func (m *manager) fixSSH(ctx context.Context) error {
 	}
 
 	for i := 0; i < 3; i++ {
-		nicName := fmt.Sprintf("%s-master%d-nic", infraID, i)
-
-		nic, err := m.interfaces.Get(ctx, resourceGroup, nicName, "")
+		// NIC names might be different if customer re-created master nodes
+		// see https://bugzilla.redhat.com/show_bug.cgi?id=188249 for more details
+		// installer naming  - <foo>-master{0,1,2}-nic
+		// machineAPI naming - <foo>-master-{0,1,2}-nic
+		nicNameInstaller := fmt.Sprintf("%s-master%d-nic", infraID, i)
+		nicNameMachineAPI := fmt.Sprintf("%s-master-%d-nic", infraID, i)
+
+		var iErr, err error
+		var nic mgmtnetwork.Interface
+		nicName := nicNameInstaller
+
+		nic, err = m.interfaces.Get(ctx, resourceGroup, nicName, "")
 		if err != nil {
-			return err
+			iErr = err // preserve original error
+			nicName = nicNameMachineAPI
+			m.log.Warnf("fallback to check MachineAPI Nic name format for %s", nicName)
+			nic, err = m.interfaces.Get(ctx, resourceGroup, nicName, "")
+			if err != nil {
+				return iErr
```

Comment:
This error will be the error from the original preserved error and not the error from the failure of the nicNameMachineAPI interface get error.  _(It will show the wrong nic in the error)_

Also, it would read easier if instead of using `nicName` in both of the functions to use `nicNameInstaller`,`nicNameMachineAPI` where you are calling the interfaces.Get

Reply:
Regarding the returned error, we'd only be in this block if `err`, and therefore `iErr` (sounds like an Apple product :), is not `nil`. Therefore, I think it's appropriate to return the original error (`iErr`).

As far as the use of `nicName`, this also appears correct to me, since it is used below this block as well, on lines 77 and 78. Setting `nicName := nicNameInstaller` as the default and then only overriding it if it can't find it is highly readable, as you can see at a glance what is the chosen value at any given point.

tl;dr: I think this is correct as is.

Reply:
Basically what Nils said :) 
We care about original error and that this is happen in the first place. Second error return would mascarade the real failure. I gonna log fallback error but not return it.

Reply:
That makes sense.  If it fails both, returning the first error is returned and logging fallback 👍🏽 .